lib/battle/CBattleInfoCallback.h

```diff
@@ -156,7 +156,7 @@ class DLL_LINKAGE CBattleInfoCallback : public virtual CBattleInfoEssentials
 	AttackableTiles getPotentiallyShootableHexes(const  battle::Unit* attacker, BattleHex destinationTile, BattleHex attackerPos) const;
 	std::vector<const battle::Unit *> getAttackedBattleUnits(const battle::Unit* attacker, BattleHex destinationTile, bool rangedAttack, BattleHex attackerPos = BattleHex::INVALID) const; //calculates range of multi-hex attacks
 	std::set<const CStack*> getAttackedCreatures(const CStack* attacker, BattleHex destinationTile, bool rangedAttack, BattleHex attackerPos = BattleHex::INVALID) const; //calculates range of multi-hex attacks
-	bool isToReverse(const battle::Unit *attacker, const battle::Unit *defender) const; //determines if attacker standing at attackerHex should reverse in order to attack defender
+	static bool isToReverse(const battle::Unit * attacker, const battle::Unit * defender); //determines if attacker standing at attackerHex should reverse in order to attack defender

```

Comment:
Same here. Only access via instance in callback

Reply:
Do not understand your static hate, but done(

Reply:
isToReverse is an interesting example.
IIRC the only reason it can be made static now is because of battle refactoring I did back in December. Before it used some other (non-static) methods from callback

Problem is, opposite is also true. It is possible that in future this function will be changed and will require access to `this` again. And any call that is not made through an instance will be broken.

Reply:
And this is not theoretical example. Back during said battle refactoring I did ended up with this particular problem - I had to make static function non-static and had a fun refactoring session in order to provide instance for every such call.

Reply:
In ideal non-static world, we can use Rust-like approach with 
```c++
struct A {
    int a;
    int b;
    int c;
    float d;
}

namespace A_impl {
    auto count_abc(A self) -> int {
    }
    auto some_other(A self) -> float {
    }
}
```

And then only operators, constructors and destructors will live inside classes.